sdk/core/azure-core/src/main/java/com/azure/core/http/HttpPipelinePosition.java

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http;
+
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.RetryPolicy;
+
+/**
+ * Indicates the position in an {@link HttpPipeline} to place an {@link HttpPipelinePolicy}.
+ */
+public enum HttpPipelinePosition {
```

Comment:
Should we call this pipeline "position" or should we name this as `HttpPolicyEvaluationStrategy`? Since this doesn't define the position but defines if this policy needs to be evaluated just once per call or evaluated for every retry attempt.

Reply:
Both would work in the scenario where we added `FIRST` and `LAST` as additional options (`HttpPipelinePosition.FIRST`, `HttpPolicyEvaluationStrategy.FIRST`). But, given this is a simple `enum` I would stick with `HttpPipelinePosition`, that way we can save `HttpPolicyEvaluationStrategy` for a more complex scenario (similar to `RetryStrategy`).

Reply:
Makes sense. Let's keep the current name then!